### PR TITLE
URLConfig documentation

### PR DESCRIPTION
**Pull request by jmosbacher**
Add some documentation for the URLConfig class

### CODE REVIEW

**Comment by jmosbacher** [comment]
Not fully complete but i guess better than nothing for the moment

File: docs/source/url_configs.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        cache=3,
+    )
+
+The details of where the model object is taken from can be determined by setting the model key of the context config
+The URL is the object being hashed, so it is important to only use pure URLs i.e the same URL should always refer to the same resource.
+
+The URL is evaluated recursively:
+  1) **tf://** - Loads a TF model from a path 
+  2) **resource://** - Loads a xenon resource by name (can also load web URLs), in this case returns a path to the file.
+  3) **cmt://** - Loads value from CMT, in this case it loads the name of the resource encoding the keras model.
```
Comment (WenzDaniel) [inline docs/source/url_configs.rst:108]: Maybe reverse order, since it is done the other way around?

File: docs/source/url_configs.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  4) **?version=ONLINE&run_id=plugin.run_id&fmt=abs_path** - Query is parsed and substituted (plugin.* are replaced with plugin attributes as evaluated at runtime) the values are then passed as keyword arguments to any protocols that include them in their signature. Everythin after the rightmost `?` character is considered the keyword arguments for the protocols.
+
+**Important** The URL arguments are sorted before they are passed to the plugin so that hashing is not sensitive to the order of the arguments.
+This is important to remember when performing tests.
```
Comment (WenzDaniel) [inline docs/source/url_configs.rst:112]: Sorted by?

**Review by WenzDaniel — APPROVED** [review]
Thanks Yossi this is already quite helpful.

File: docs/source/url_configs.rst (hunk 2 above)
Comment (jmosbacher) [reply docs/source/url_configs.rst:112]: I am assuming the PR #827 will be merged at some point...

File: docs/source/url_configs.rst (hunk 1 above)
Comment (jmosbacher) [reply docs/source/url_configs.rst:108]: yeah i guess that would make more sense

**Pull request merged**